Makefile

```diff
@@ -61,6 +61,8 @@ test-functional:  WHAT = ./tests/...
 test-functional:
 	./hack/build/run-functional-tests.sh ${WHAT} "${TEST_ARGS}"
 
+test-functional-ci: build-functest test-functional
```

Comment:
Why do we have to explicitly build these?  I expected this to happen automatically.  Is this somehow different from the developer workflow?  Typically make is supposed to handle build dependencies for you but that is probably not working since we rely on so many .PHONY targets.

Reply:
The reason is to not pollute the developer work flow. In the developer workflow we have go available and thus can call go test .... however in CI go is not available, so we have to run build-functest (maybe name that something else) that builds the test.tests executable, that is executed in the CI workflow. The only difference between a local test-functional and a test-functional-ci is the build of the executable.